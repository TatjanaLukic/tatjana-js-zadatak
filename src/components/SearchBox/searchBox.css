@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;0,500;0,700;1,400&display=swap");
.searchBox {
  display: block;
  box-sizing: border-box;
  font-family: "Roboto", sans-serif;
  height: 3.5rem;
  width: 90%;
  margin: auto;
  margin-top: 1rem;
  background-color: rgba(221, 221, 221, 0.3);
  border: none;
  font-size: 1rem;
  padding: 1rem;
  font-size: 1.4rem;
}
@media screen and (min-width: 576px) {
  .searchBox {
    width: 50%;
    margin: auto;
    margin-top: 1rem;
    height: 4rem;
  }
}
@media screen and (min-width: 992px) {
  .searchBox {
    width: 50%;
    margin: auto;
    margin-top: 2rem;
  }
}

/*# sourceMappingURL=searchBox.css.map */
