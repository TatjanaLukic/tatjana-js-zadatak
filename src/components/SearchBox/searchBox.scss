@import "../../assets/variables";
.searchBox {
  display: block;
  box-sizing: border-box;
  font-family: 'Roboto', sans-serif;
  height: 3.5rem;
  width: 90%;
  margin: auto;
  margin-top: 1rem;
  background-color:rgba($news-tile-border,0.3);
  border: none;
  font-size: 1rem;
  padding: 1rem;
  font-size: 1.4rem;
 

  @media screen and (min-width: $smallSize) {
    width: 50%;
    margin: auto;
    margin-top: 1rem;
    height: 4rem;
  }

  @media screen and (min-width: $largeSize) {
    width: 50%;
    margin: auto;
    margin-top: 2rem;
}

}
