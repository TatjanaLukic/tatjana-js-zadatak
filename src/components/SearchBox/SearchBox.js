import React from "react";
import "./searchBox.css";

const SearchBox = ({term, onChangeHandler}) => {

  return (
    <input
      className="searchBox"
      type="search"
      value={term}
      autoFocus={true}
      placeholder="Search term..."
      onChange={onChangeHandler}  
    />
  );
};

export default SearchBox;
