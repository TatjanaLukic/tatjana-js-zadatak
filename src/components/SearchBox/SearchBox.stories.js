import React from "react";
import SearchBox from "./SearchBox";

export default {
  title: "App/SearchBox",
  component: SearchBox

};

const mockFn = ()=>{};
const Template = (args) => <SearchBox {...args} />;

export const Default = Template.bind({});
Default.args = {
term:"Search term",
onChangeHandler: {mockFn}
};
