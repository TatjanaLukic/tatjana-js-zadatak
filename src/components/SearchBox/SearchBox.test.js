import React from "react";
import { shallow,mount } from "enzyme";
import SearchBox from "./SearchBox";

describe("<SearchBox />", () => {
  let searchBox;
  const fn = () => {};

  it("should have input", () => {
    searchBox = mount(<SearchBox term="Search term" onChangeHandler={fn} />);
    expect(searchBox.find('input')).toHaveLength(1);
    searchBox.unmount();
  });
  it("spanshot matches", () => {
    searchBox = shallow(<SearchBox term="Search term" onChangeHandler={fn} />);
    expect(searchBox).toMatchSnapshot();
  });
});
