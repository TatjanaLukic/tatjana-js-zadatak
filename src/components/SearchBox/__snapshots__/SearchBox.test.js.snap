// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> spanshot matches 1`] = `
<input
  autoFocus={true}
  className="searchBox"
  onChange={[Function]}
  placeholder="Search term..."
  type="search"
  value="Search term"
/>
`;
